library/IvozProvider/Mapper/Sql/RetailAccounts.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * Application Model Mapper
+ *
+ * @package Mapper
+ * @subpackage Sql
+ * @author Luis Felipe Garcia
+ * @copyright ZF model generator
+ * @license http://framework.zend.com/license/new-bsd     New BSD License
+ */
+
+/**
+ * Data Mapper implementation for IvozProvider\Model\RetailAccounts
+ *
+ * @package Mapper
+ * @subpackage Sql
+ * @author Luis Felipe Garcia
+ */
+namespace IvozProvider\Mapper\Sql;
+class RetailAccounts extends Raw\RetailAccounts
+{
+    /**
+     * Saves current row
+     * @return integer primary key for autoincrement fields if the save action was successful
+     */
+    protected function _save(\IvozProvider\Model\Raw\RetailAccounts $model,
+        $recursive = false, $useTransaction = true, $transactionTag = null, $forceInsert = false
+    ) {
+        // Set account domain to its brand domain
+        $model->setDomain($model->getCompany()->getBrand()->getDomainUsers());
+
+        $response = parent::_save($model, $recursive, $useTransaction, $transactionTag, $forceInsert);
+        if ($response) {
+            // Replicate Account into ast_ps_endpoint
+            $endpointMapper = new \IvozProvider\Mapper\Sql\AstPsEndpoints();
+            $endpoint = $endpointMapper->findOneByField("retailAccountId", $response);
+
+            // If not found create a new one
+            $forceInsert = false;
+            if (is_null($endpoint)) {
+                $forceInsert = true;
+                $endpoint = new \IvozProvider\Model\AstPsEndpoints();
+                $endpoint->setContext("retail")
+                    ->setSendDiversion("yes")
+                    ->setSendPai("yes");
+            }
+
+            // Update/Insert endpoint data
+            $endpoint->setRetailAccountId($response)
+                ->setSorceryId($model->getSorcery())
+                ->setFromDomain($model->getDomain())
+                ->setAors($model->getSorcery())
+                ->setDisallow($model->getDisallow())
+                ->setAllow($model->getAllow())
+                ->setDirectmediaMethod($model->getDirectmediaMethod())
+                ->setTrustIdInbound('yes')
+                ->setOutboundProxy('sip:users.ivozprovider.local^3Blr')
```

Comment:
There's another approach: Never set it and make it the default value for this database field.
```
`outbound_proxy` varchar(256) DEFAULT NULL,
```
```
`outbound_proxy` varchar(256) NOT NULL DEFAULT 'sip:users.ivozprovider.local^3Blr';
```